### PR TITLE
Nexus 5418 store checksum in item attributes

### PR DESCRIPTION
**Pull request by ifedorenko**
I could not figure out how to convince bamboo to build this branch :-(

### CODE REVIEW

**Comment by ifedorenko** [comment]
Bamboo is happy https://bamboo.zion.sonatype.com/browse/NXO-OSSF5-2

File: nexus-core/src/main/java/org/sonatype/nexus/proxy/maven/ChecksumContentValidator.java [hunk 1]
```diff
@@ -101,92 +131,155 @@ protected RemoteHashResponse retrieveRemoteHash( AbstractStorageItem item, Proxy
...
+            {
+                try
+                {
+                    final StorageFileItem remoteItem =
+                        (StorageFileItem) proxy.getRemoteStorage().retrieveItem( proxy, request, proxy.getRemoteUrl() );
+                    hash = MUtils.readDigestFromFileItem( remoteItem ); // closes http input stream
+                }
+                catch ( ItemNotFoundException e )
+                {
+                    // fall through
```
Comment (jdillon) [inline nexus-core/src/main/java/org/sonatype/nexus/proxy/maven/ChecksumContentValidator.java:222]: What do you mean by "fall through" here, I'm used to seeing this in a switch block, but not in a catch block.

Comment (ifedorenko) [inline nexus-core/src/main/java/org/sonatype/nexus/proxy/maven/ChecksumContentValidator.java:222]: The exception is intentionally ignored and the code below is expected to recover/tolerate it.

**Comment by cstamas** [comment]
+1

**Pull request merged**